src/output/help_template.rs

```diff
@@ -816,6 +816,20 @@ impl<'cmd, 'writer> HelpTemplate<'cmd, 'writer> {
     fn spaces(&mut self, n: usize) {
         self.none(self.get_spaces(n));
     }
+
+    fn should_show_arg(&self, arg: &Arg) -> bool {
```

Comment:
> This cuts out a large fraction of the references to `use_long`,
making it easier to see the logic of how it gets used.

As you mentioned, it can help in finding uses but it also hurts in that you can't tell by looking at a call what all state its dependent on.  In looking at these other changes and evaluating them, I'm finding the latter more of a problem.

I would prefer this commit to be reverted.